### ISSUE TITLE
Cannot update challenge_protocol, bis

### ISSUE THREAD

**Issue by tech-kio** [zopefoundation/Products.PluggableAuthService#66]
Hello,

As a follow-up on #63, using Zope 4.2.1, PAS still has problems selecting http auth for XML-RPC in acl_users/challenge_protocol.

It doesn't save changes after pressing update mapping.

1- if I have the following ordering ; (any), cookie_auth, http, I can only save cookie_auth. After pressing update mapping, it doesn't keep the setting for the other one.

2- if you go in acl_users/plugins/Challenge Plugins and you change the ordering on the right, between basic_auth and cookie auth, all selections in acl_users/challenge_protocol are lost!

Thank you for your help!

**Comment by jugmac00** [comment]
Which version of PAS are you using?

**Comment by tech-kio** [comment]
Products.PluggableAuthService 2.4

**Comment by icemac** [comment]
@dataflake Do you have any idea what might cause this issue? (I am not sure why I have the feeling that there was something similar which got already fixed.)

**Comment by dataflake** [comment]
I'll look at it when I have time.

**Comment by dataflake** [comment]
Your explanations are hard to follow because you don't explain which specific plugins and tabs you are referring to. I am attempting to guess:

1 - I am guessing you mean the "Mapping" tab on the Challenge Protocol Chooser plugin. I cannot reproduce your issue there. I can select any combination, click "Update Mapping" and it will stick.

2. I am guessing you clicked into the Plugin Registry and then "Challenge Plugins" on the "Plugins" ZMI tab and then you change the order of the two plugins in the box on the right. I have followed these steps and when I return to the "Mapping" tab on the Challenge Protocol Chooser plugin I still see the same mappings.

Changing the order of plugins inside the Plugin Registry has nothing to do with the plugins you select in the Challenge Protocol Chooser mappings. Those mappings are only affected if you deactivate a plugin in the Plugin Registry by moving it from the right hand side box to the left.

I cannot reproduce the issue at all.

**Comment by tech-kio** [comment]
So sorry, my mistake. It is working fine now.
I have a custom version and forgot to sync the v2.4 file : plugins/ChallengeProtocolChooser.py
namely from:
            value = filter(None, value)
to:
            value = [_f for _f in value if _f]

Everything is fine now. Thank you very much. You can close.
